R-package/R/lgb.Dataset.R

```diff
@@ -32,6 +32,10 @@ Dataset <- R6::R6Class(
                           free_raw_data = TRUE,
                           used_indices = NULL,
                           info = list(),
+                          label = NULL,
+                          weight = NULL,
+                          init_score = NULL,
+                          group = NULL,
```

Comment:
Is this change and the corresponding changes below necessary to support the feature "Accept data frames as inputs"? Or do you just have a personal preference for explicit keyword arguments instead of the `info` construct that exists in this class today?

If it's not strictly required to support this feature, please revert this change, make this PR work with the existing interface of `lgb.Dataset`, and open an issue describing what you'd like to change about the interface to `lgb.Dataset()` and how you think that proposed change improves LightGBM.

That will reduce the size of the diff here, which should allow us to provide a higher-quality review and bring this PR to a resolution more quickly.

Reply:
It's necessary, as otherwise it wouldn't be possible to pass them as column names.

Reply:
> otherwise it wouldn't be possible to pass them as column names.

Ok, please revert any changes related to being able to pass these characteristics as column names. I'm happy to have a discussion about that in a separate feature request issue, but I don't think that change is strictly required to accomplish the behavior "accept data frames as inputs".

We have a strong preference in this project for incremental progress through as-small-as-possible pull requests focused on a single thing. That enables the team of maintainers here (mostly volunteers) to provide higher-quality reviews and prevents the situation where pull requests touching a lot of code drag on for a long time and block or complicate other development.